sympy/assumptions/tests/test_smtask.py

```diff
@@ -0,0 +1,19 @@
+from sympy.assumptions.smtask import z3ask
+from sympy.core.symbol import symbols
+from sympy.testing.pytest import raises, XFAIL
+from sympy.assumptions.ask import Q
+x, y, z = symbols('x y z')
+def test_z3ask():
+    assert z3ask(y > 0, (y < -x**2 +.5)) is None
+    assert z3ask(y > 0, (y < -x**2 +.5) & Q.integer(y)) is False
+    assert z3ask(x > z, (x > y) & (y > z)) is True
+
+
+@XFAIL
+def test_z3ask_fail():
+    assert z3ask((y > 0), (y > 2) & Q.imaginary(y)) is False # currently returns True
+    raises(ValueError, lambda:  z3ask((y > 0), (y > 2) & Q.imaginary(y)))
+    assert z3ask((y > 0) & Q.imaginary(y), (y > 2))  is False # currently returns True
```

Comment:
Let's also make sure we support the cases where Z3 can't handle the predicate at all. It should error rather than giving a wrong answer. If there aren't any such predicates, we can create a fake one.

Reply:
My changes to the SMT-Lib printer make it so that any predicate that's not handled is converted to a boolean dummy variable so there shouldn't be any need to throw an error anymore.